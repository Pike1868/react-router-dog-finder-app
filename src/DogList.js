import React from "react";
import { Link } from "react-router-dom";

/**
 * DogList Component:
 *
 * Returns each dog displayed in a div, with its image, and name as a Link to that dogs detail page
 *
 * Props:
 * - dogs (array): Array of dog objects
 *
 */

const DogList = ({ dogs }) => {
  return (
    <div className="DogList">
      <div className="row mt-4">
        <div className="col">
          <h2 className="text-center">
            HELLO. WE HAVE DOGS. CLICK ON THEM FOR MORE INFO.
          </h2>
        </div>
      </div>
      <div className="row">
        {dogs.map((d) => (
          <div className="col-3 text-center" key={d.name}>
            <img src={`/${d.src}.jpg`} alt={d.name} />
            <h3 className="mt-3">
              <Link to={`/dogs/${d.name.toLowerCase()}`}>{d.name}</Link>
            </h3>
          </div>
        ))}
      </div>
    </div>
  );
};

export default DogList;
