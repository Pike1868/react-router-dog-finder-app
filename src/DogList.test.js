import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import DogList from "./DogList";
import dogs from "./dogs";

//Smoke Test
test("DogList component renders without crashing", () => {
  render(
    <MemoryRouter>
      <DogList dogs={dogs} />
    </MemoryRouter>
  );
});

//Snapshot Test
test("DogList component matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <DogList dogs={dogs} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

describe("DogList Component Tests", () => {
  beforeEach(() => {
    render(
      <MemoryRouter>
        <DogList dogs={dogs} />
      </MemoryRouter>
    );
  });

  test("displays the heading message", () => {
    const headingMsg = screen.getByText(
      "HELLO. WE HAVE DOGS. CLICK ON THEM FOR MORE INFO."
    );
    expect(headingMsg).toBeInTheDocument();
  });

  //Dynamic dog link test
  dogs.forEach((dog) => {
    test(`displays a link for ${dog.name}`, () => {
      const whiskeyLink = screen.getByText("Whiskey");
      expect(whiskeyLink).toBeInTheDocument();
    });
  });

  test("displays images with correct alt text", () => {
    dogs.forEach((dog) => {
      const image = screen.getByAltText(dog.name);
      expect(image).toBeInTheDocument();
    });
  });
});
