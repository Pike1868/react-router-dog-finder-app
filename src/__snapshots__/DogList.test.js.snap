// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DogList"
  >
    <div
      class="row mt-4"
    >
      <div
        class="col"
      >
        <h2
          class="text-center"
        >
          HELLO. WE HAVE DOGS. CLICK ON THEM FOR MORE INFO.
        </h2>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-3 text-center"
      >
        <img
          alt="Whiskey"
          src="/whiskey.jpg"
        />
        <h3
          class="mt-3"
        >
          <a
            href="/dogs/whiskey"
          >
            Whiskey
          </a>
        </h3>
      </div>
      <div
        class="col-3 text-center"
      >
        <img
          alt="Duke"
          src="/duke.jpg"
        />
        <h3
          class="mt-3"
        >
          <a
            href="/dogs/duke"
          >
            Duke
          </a>
        </h3>
      </div>
      <div
        class="col-3 text-center"
      >
        <img
          alt="Perry"
          src="/perry.jpg"
        />
        <h3
          class="mt-3"
        >
          <a
            href="/dogs/perry"
          >
            Perry
          </a>
        </h3>
      </div>
      <div
        class="col-3 text-center"
      >
        <img
          alt="Tubby"
          src="/tubby.jpg"
        />
        <h3
          class="mt-3"
        >
          <a
            href="/dogs/tubby"
          >
            Tubby
          </a>
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
